### PR TITLE
Fix tests failing due to whitespaces and trailing forward slash in Advanced Node and Express

### LINKED ISSUE #39533: Advanced Node and Express - Trailing Forward Slash and Whitespaces Fails Tests
**Describe your problem and how to reproduce it:**
Trailing forward slash and whitespaces fails the tests

**Add a Link to the page with the problem:**
- [Implementation of Social Authentication](https://www.freecodecamp.org/learn/quality-assurance/advanced-node-and-express/implementation-of-social-authentication)

**Tell us about your browser and operating system:**
* Browser Name: Chrome 84
* Browser Version: Chrome/84.0.4147.135
* Operating System: Windows 10

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
image

### PR DESCRIPTION
**Pull request by piouson**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39533

### CODE REVIEW

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md [hunk 1]
```diff
@@ -40,9 +40,9 @@ Submit your page when you think you've got it right. If you're running into erro
 ```yml
 tests:
   - text: Route /auth/github should be correct.
-    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data, /('|")\/auth\/github('|")[^]*get.*passport.authenticate.*github/gi, 'Route auth/github should only call passport.authenticate with github'); }, xhr => { throw new Error(xhr.statusText); })
+    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data.replace(/[\s\r\n]+/g,""), /('|")\/auth\/github\/?('|")[^]*?get.*?passport.authenticate.*?github/gi, 'Route auth/github should only call passport.authenticate with github'); }, xhr => { throw new Error(xhr.statusText); })
```
Comment (Sky020) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md:43]:
```suggestion
    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data.replace(/\s/g,""), /('|")\/auth\/github\/?('|")[^]*?get.*?passport.authenticate.*?github/gi, 'Route auth/github should only call passport.authenticate with github'); }, xhr => { throw new Error(xhr.statusText); })
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md [hunk 2]
```diff
@@ -40,9 +40,9 @@ Submit your page when you think you've got it right. If you're running into erro
...
   - text: Route /auth/github/callback should be correct.
-    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data, /('|")\/auth\/github\/callback('|")[^]*get.*passport.authenticate.*github.*failureRedirect:( |)("|')\/\2/gi, 'Route auth/github/callback should accept a get request and call passport.authenticate for github with a failure redirect to home'); }, xhr => { throw new Error(xhr.statusText); })
+    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data.replace(/[\s\r\n]+/g,""), /('|")\/auth\/github\/callback\/?('|")[^]*?get.*?passport.authenticate.*?github.*?failureRedirect:("|')\/\2/gi, 'Route auth/github/callback should accept a get request and call passport.authenticate for github with a failure redirect to home'); }, xhr => { throw new Error(xhr.statusText); })
```
Comment (Sky020) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md:45]:
```suggestion
    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data.replace(/\s/g,""), /('|")\/auth\/github\/callback\/?('|")[^]*?get.*?passport.authenticate.*?github.*?failureRedirect:("|')\/\2/gi, 'Route auth/github/callback should accept a get request and call passport.authenticate for github with a failure redirect to home'); }, xhr => { throw new Error(xhr.statusText); })
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I have not tested the need for the lazy matching, but the `\s` flag checks for all kinds of whitespace (`\r\n\t\f\v`), and the `g` flag should negate the need for `+`

**Comment by RandellDawson** [comment]
@Sky020 If we are fixing the regex, can you adjust the regex for both of your suggestions to make use of the capture groups for the single or double quotes.  For example:
Change:
```js
/('|")\/auth\/github\/callback\/?('|")[^]*?get.*?passport.authenticate.*?github.*?failureRedirect:("|')\/\2/gi
```
to:
```js
/('|")\/auth\/github\/callback\/?\1[^]*?get.*?passport.authenticate.*?github.*?failureRedirect:("|')\/\2/gi
```

**Comment by piouson** [comment]
> I have not tested the need for the lazy matching, but the `\s` flag checks for all kinds of whitespace (`\r\n\t\f\v`), and the `g` flag should negate the need for `+`

I suppose only the first test requires the lazy matching, but I left both lazy..

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md [hunk 3]
```diff
@@ -40,9 +40,9 @@ Submit your page when you think you've got it right. If you're running into erro
...
   - text: Route /auth/github/callback should be correct.
-    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data, /('|")\/auth\/github\/callback('|")[^]*get.*passport.authenticate.*github.*failureRedirect:( |)("|')\/\2/gi, 'Route auth/github/callback should accept a get request and call passport.authenticate for github with a failure redirect to home'); }, xhr => { throw new Error(xhr.statusText); })
+    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data.replace(/\s/g,""), /('|")\/auth\/github\/callback\/?\1[^]*?get.*?passport.authenticate.*?github.*?failureRedirect:\1\/\1/gi, 'Route auth/github/callback should accept a get request and call passport.authenticate for github with a failure redirect to home'); }, xhr => { throw new Error(xhr.statusText); })
```
Comment (Sky020) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md:45]:
```suggestion
    testString: getUserInput => $.get(getUserInput('url')+ '/_api/routes.js') .then(data => { assert.match(data.replace(/\s/g,""), /('|")\/auth\/github\/callback\/?\1[^]*?get.*?passport.authenticate.*?github.*?failureRedirect:( |)("|')\/\2/gi, 'Route auth/github/callback should accept a get request and call passport.authenticate for github with a failure redirect to home'); }, xhr => { throw new Error(xhr.statusText); })
```

Comment (Sky020) [reply curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.english.md:45]: Unlikely, but we cannot assume the camper will always use the same kind of quote.

**Comment by Sky020** [comment]
@piouson Thank you, for making those changes. Unfortunately, I had a brain-freeze moment. So, could you update to remove the `( |)` again?

You were correct in removing it in the first place, because of the `.replace()`

**Comment by RandellDawson** [comment]
@Sky020 Technically, the second set of quotes around `/` can be either `"` or `'`.  While not likely, the user could use double quotes for one and single quotes for the other.

**Comment by piouson** [comment]
> @Sky020 Technically, the second set of quotes around `/` can be either `"` or `'`. While not likely, the user could use double quotes for one and single quotes for the other.

ahah I realise now, best to leave it as separate capture groups..

> @piouson Thank you, for making those changes. Unfortunately, I had a brain-freeze moment. So, could you update to remove the `( |)` again?
> 
> You were correct in removing it in the first place, because of the `.replace()`

good catch, indeed that was my initial idea, I missed that, but all should be good now, thank you!

**Review by RandellDawson — APPROVED** [review]
LGTM

**Review by ShaunSHamilton — APPROVED** [review]
Thank you, @piouson ! 👍
We are happy to accept these changes, and look forward to future contributions. 🎉

**Review by moT01 — APPROVED** [review]
Thanks for contributing to freeCodeCamp @piouson 🎉

**Pull request merged**